spec/controllers/admin/admin_users_controller_spec.rb

```diff
@@ -236,6 +268,68 @@
           end
         end
       end
+
+      context "with available user roles restricted by type of admin" do
+        let!(:opendoors_role) { create(:role, name: "opendoors") }
+        let!(:archivist_role) { create(:role, name: "archivist") }
+        let!(:tag_wrangler_role) { create(:role, name: "tag_wrangler") }
+
+        before do
+          admin.update!(roles: ["open_doors"])
+        end
+
+        it "can add a permitted role" do
+          user.update!(roles: [])
+
+          put :update, params: {
+            id: user.login,
+            user: {
+              roles: [opendoors_role.id.to_s]
+            }
+          }
+          expect(flash[:notice]).to eq "User was successfully updated."
+          expect(user.reload.roles).to eq([opendoors_role])
+        end
+
+        it "can remove a permitted role" do
+          user.update!(roles: [opendoors_role, archivist_role])
+
+          put :update, params: {
+            id: user.login,
+            user: {
+              roles: [opendoors_role.id.to_s]
+            }
+          }
+          expect(flash[:notice]).to eq "User was successfully updated."
+          expect(user.reload.roles).to eq([opendoors_role])
+        end
+
+        it "does not add unpermitted roles" do
+          user.update!(roles: [opendoors_role])
+
+          put :update, params: {
+            id: user.login,
+            user: {
+              roles: [opendoors_role.id.to_s, tag_wrangler_role.id.to_s]
+            }
+          }
+          expect(flash[:notice]).to eq "User was successfully updated."
+          expect(user.reload.roles).to eq([opendoors_role])
+        end
+
+        it "does not remove unpermitted roles" do
+          user.update!(roles: [opendoors_role, tag_wrangler_role])
```

Comment:
Would it be better if the user was just starting with `tag_wrangler_role`, since we're adding the `opendoors_role`?

Relatedly, could we have a test to verify the user retains their `tag_wrangling_role` if they started off with `[opendoors_role, tag_wrangler_role]` and our OD admin _removes_ `opendoors_role`?